src/app/components/user_details/user_details.rs

```diff
@@ -45,15 +47,20 @@ impl UserDetails {
 
         if let Some(store) = model.get_list_store() {
             let weak_model = Rc::downgrade(&model);
-            let worker_clone = worker.clone();
 
             widget
                 .user_playlists
                 .bind_model(Some(store.unsafe_store()), move |item| {
-                    let item = item.downcast_ref::<AlbumModel>().unwrap();
-                    let child = create_album_for(item, worker_clone.clone(), weak_model.clone());
-                    child.show_all();
-                    child.upcast::<gtk::Widget>()
+                    wrap_flowbox_item(item, |item: &AlbumModel| {
+                        let album = Album::new(item, worker.clone());
+                        let weak = weak_model.clone();
+                        album.connect_album_pressed(move |a| {
```

Comment:
Sorry, last one :p clone! works here as well

Reply:
:facepalm: lol can't believe I did that